lib/metanorma/compile_options.rb

```diff
@@ -0,0 +1,83 @@
+require "csv"
+
+module Metanorma
+  class Compile
+    def require_libraries(options)
+      options&.dig(:require)&.each { |r| require r }
+    end
+
+    def xml_options_extract(file)
+      xml = Nokogiri::XML(file) { |config| config.huge }
+      if xml.root
+        @registry.root_tags.each do |k, v|
+          return { type: k } if v == xml.root.name
+        end
+      end
+      {}
+    end
+
+    def options_extract(filename, options)
+      content = read_file(filename)
+      o = Metanorma::Input::Asciidoc.new.extract_metanorma_options(content)
+        .merge(xml_options_extract(content))
+      options[:type] ||= o[:type]&.to_sym
+      t = @registry.alias(options[:type]) and options[:type] = t
+      dir = filename.sub(%r(/[^/]+$), "/")
+      options[:relaton] ||= "#{dir}/#{o[:relaton]}" if o[:relaton]
+      options[:sourcecode] ||= "#{dir}/#{o[:sourcecode]}" if o[:sourcecode]
```

Comment:
I propose to use `File.join` instead of `{p1}/{p2}` approach, many times we faced with issues related to path concatenation especially on windows with absolute path

Reply:
Fully agree. We must use platform independent methods.